In [1]:
#import necessary library
import pandas as pd
import requests
import csv
from urllib.request import Request, urlopen
from bs4 import BeautifulSoup #For data scrapping

In [2]:
#For mounting the drive
#If you want to run it on your local machine then skip it and change the csv diractory as your own interest
from google.colab import drive
drive.mount('/content/gdrive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/gdrive


In [3]:
#Html Request to the website.
page = requests.get("https://www.worldometers.info/coronavirus/?fbclid=IwAR0L9-HT-jDr8yXdZaeIsKT_kLkwFyGxwGGa8ezWQ2lGRrDd4Kjk_zqWtE8#news")
soup = BeautifulSoup(page.content,'html.parser')
#from the website, All we need is the table.
table = soup.findAll("table", {"id":"main_table_countries_yesterday"})[0]
# Fatch all the rows and store in a list.
rows = table.findAll("tr")

# Store as a CSV format

with open("/content/gdrive/My Drive/Colab Notebooks/COVID-19/covid_update2.csv", "wt+",encoding="utf-8", newline="") as f:
    writer = csv.writer(f)
    for row in rows:
      csv_row = []
      for cell in row.findAll(["td", "th"]):
        csv_row.append(cell.get_text())
      writer.writerow(csv_row)


# Data Exploration

In [4]:
#importing necessary library
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
sns.set() # Apply the default seaborn theme, scaling, and color palette

import pandas as pd
import numpy as np

import warnings
warnings.filterwarnings("ignore")

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [20]:
df1 = pd.read_csv("/content/gdrive/My Drive/Colab Notebooks/COVID-19/covid_update.csv")
df2 = pd.read_csv("/content/gdrive/My Drive/Colab Notebooks/COVID-19/covid_update2.csv")

In [21]:
cols1 = df1.columns
cols2 = df2.columns

In [22]:
flag = 0
for col in cols1:
  if col in cols2:
    df1[col] = df2[col]
df = df1


In [9]:
df.head(10)

#      Country,Other  ... 1 Deathevery X ppl 1 Testevery X ppl
0  NaN           \nAsia\n  ...                NaN               NaN
1  NaN  \nNorth America\n  ...                NaN               NaN
2  NaN  \nSouth America\n  ...                NaN               NaN
3  NaN         \nEurope\n  ...                NaN               NaN
4  NaN         \nAfrica\n  ...                NaN               NaN
5  NaN        \nOceania\n  ...                NaN               NaN
6  NaN               \n\n  ...                NaN               NaN
7  NaN              World  ...                NaN               NaN
8  1.0              China  ...            310,601                16
9  2.0                USA  ...              2,403                 8

[10 rows x 19 columns]

At first we need to name our features in a meaningful and authentic way. 

In [23]:
df.columns = ['#','Country','TotalCases','NewCases','TotalDeaths','NewDeaths','TotalRecovered','NewRecovered','ActiveCases','SeriousCritical','TotalCasesPerMillion','DeathsPerMillion','TotalTests','TestsPerMillion','Population','Continent','CaseEvery','DeathEvery','TestEvery']
df.head()

#            Country TotalCases  ... CaseEvery DeathEvery TestEvery
0 NaN           \nAsia\n  2,959,505  ...        \n        NaN       NaN
1 NaN  \nNorth America\n  3,995,688  ...        \n        NaN       NaN
2 NaN  \nSouth America\n  2,893,624  ...        \n        NaN       NaN
3 NaN         \nEurope\n  2,570,985  ...        \n        NaN       NaN
4 NaN         \nAfrica\n    598,236  ...        \n        NaN       NaN

[5 rows x 19 columns]

We've seen some columns are not that much rellivent. We'll reduce those columns

In [24]:
df = df.drop(['#','CaseEvery','DeathEvery','TestEvery'], axis=1)
df.head(10)

Country  TotalCases  ...      Population          Continent
0           \nAsia\n   2,959,505  ...             NaN               Asia
1  \nNorth America\n   3,995,688  ...             NaN      North America
2  \nSouth America\n   2,893,624  ...             NaN      South America
3         \nEurope\n   2,570,985  ...             NaN             Europe
4         \nAfrica\n     598,236  ...             NaN             Africa
5        \nOceania\n      11,460  ...             NaN  Australia/Oceania
6               \n\n         721  ...             NaN                NaN
7              World  13,030,219  ...             NaN                All
8              China      83,594  ...  1,439,323,776                Asia
9                USA   3,413,995  ...    331,065,797       North America

[10 rows x 15 columns]

The First 8 Rows data is continantal data. We will deal with only countries data. So We're going to delete first 8 row from this dataset. 

In [25]:
df = df.iloc[8:]
df.head()


Country TotalCases NewCases  ... TestsPerMillion      Population      Continent
8    China     83,594       +7  ...          62,814  1,439,323,776            Asia
9      USA  3,413,995  +58,349  ...         128,281    331,065,797   North America
10  Brazil  1,866,176  +25,364  ...          21,508    212,607,642   South America
11   India    879,466  +29,108  ...           8,394  1,380,418,989            Asia
12  Russia    727,162   +6,615  ...         157,815    145,936,664          Europe

[5 rows x 15 columns]

We can see the index start from 8. So we're going to reset the index number.

In [26]:
df = df.reset_index(drop=True)
df.head()

Country TotalCases NewCases  ... TestsPerMillion      Population      Continent
0   China     83,594       +7  ...          62,814  1,439,323,776            Asia
1     USA  3,413,995  +58,349  ...         128,281    331,065,797   North America
2  Brazil  1,866,176  +25,364  ...          21,508    212,607,642   South America
3   India    879,466  +29,108  ...           8,394  1,380,418,989            Asia
4  Russia    727,162   +6,615  ...         157,815    145,936,664          Europe

[5 rows x 15 columns]

In [14]:
#Data discription
df.describe()

Country TotalCases NewCases  ... TestsPerMillion Population Continent
count      223        223      144  ...             194        215       220
unique     216        221      114  ...             194        214         7
top     Total:         12       +3  ...         128,281               Africa
freq         8          2        5  ...               1          2        58

[4 rows x 15 columns]

From this Description We got some interesting instinct. <br>
### 1: Tatal affected countries = 215 
### 2: Most affectected Continent = Africa
### 3: Total affect countries in Africa = 58

In [15]:
df.shape

(223, 15)

In [16]:
df.tail(10)

Country  TotalCases  ... Population          Continent
213               Anguilla           3  ...    15,007       North America
214  Saint Pierre Miquelon           2  ...     5,793       North America
215                 Total:   2,959,505  ...        NaN               Asia
216                 Total:   3,995,688  ...        NaN      North America
217                 Total:   2,893,624  ...        NaN      South America
218                 Total:   2,570,985  ...        NaN             Europe
219                 Total:     598,236  ...        NaN             Africa
220                 Total:      11,460  ...        NaN  Australia/Oceania
221                 Total:         721  ...        NaN                NaN
222                 Total:  13,030,219  ...        NaN                All

[10 rows x 15 columns]

In [27]:
row_size, col_size = df.shape

In [28]:

df = df.drop(df.index[row_size-8:row_size-1])
df = df.reset_index(drop=True)
df.tail()

Country  TotalCases  ... Population      Continent
211  Caribbean Netherlands           7  ...    26,231   North America
212              St. Barth           6  ...     9,878   North America
213               Anguilla           3  ...    15,007   North America
214  Saint Pierre Miquelon           2  ...     5,793   North America
215                 Total:  13,030,219  ...        NaN            All

[5 rows x 15 columns]

In [30]:
df.fillna(0, inplace=True)
df.tail()

Country  TotalCases  ... Population      Continent
211  Caribbean Netherlands           7  ...    26,231   North America
212              St. Barth           6  ...     9,878   North America
213               Anguilla           3  ...    15,007   North America
214  Saint Pierre Miquelon           2  ...     5,793   North America
215                 Total:  13,030,219  ...          0            All

[5 rows x 15 columns]

In [44]:
total_null = df[df.Continent==0].index.tolist() + df[df.Country==0].index.tolist()
total_null = list(set(total_null))
df.drop(df.index[total_null], inplace=True)
df.shape

(214, 15)

In [53]:
df = df[df.columns].replace({'\+':''}, regex = True)
df.tail()



Country  TotalCases  ... Population      Continent
211  Caribbean Netherlands           7  ...    26,231   North America
212              St. Barth           6  ...     9,878   North America
213               Anguilla           3  ...    15,007   North America
214  Saint Pierre Miquelon           2  ...     5,793   North America
215                 Total:  13,030,219  ...          0            All

[5 rows x 15 columns]

In [54]:
df = df.replace(' ', 0)
df.tail()
                           
                           

Country  TotalCases  ... Population      Continent
211  Caribbean Netherlands           7  ...    26,231   North America
212              St. Barth           6  ...     9,878   North America
213               Anguilla           3  ...    15,007   North America
214  Saint Pierre Miquelon           2  ...     5,793   North America
215                 Total:  13,030,219  ...          0            All

[5 rows x 15 columns]